### PR TITLE
Study view: support old chart Ids when retrieving user preference settings

### PR DESCRIPTION
**Pull request by kalletlak**
Fix https://github.com/cBioPortal/cbioportal/issues/7046

CC: @jjgao

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -2512,23 +2512,31 @@ export class StudyViewPageStore {
     private loadSettings(chartSettngs: ChartUserSetting[]){
         this.clearPageSettings();
         _.map(chartSettngs, chartUserSettings => {
+            const regex = /^(?:PATIENT_|SAMPLE_)([a-zA-Z0-9_]+)/i;
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: Let's move this to study  view util.  Also the similar logic converting the chart id to clinical attribute should be put together(getClinicalDataBySamples method)

Comment (kalletlak) [reply src/pages/studyView/StudyViewPageStore.ts]: I don't see any similarity between getClinicalDataBySamples and chart id. Do you mean to move getClinicalDataBySamples method to utils?

File: src/pages/studyView/StudyViewUtils.tsx [hunk 2]
```diff
@@ -2417,3 +2418,102 @@ export function getFilteredAndCompressedDataIntervalFilters(
...
+        ) {
+            numberOfClinicalAttributeCharts++;
+            let match = chartSetting.id.match(chartIdWithDataTypeRegex);
+            if (!!match) {
+                numberOfChartRequiringUpdates++;
+            }
+        }
+    });
+
+    // Only update ids if all the chart ids require updates.
```
Comment (alisman) [inline src/pages/studyView/StudyViewUtils.tsx]: this comment could use further elaboration.  i just don't really understand business logic here.

File: src/pages/studyView/StudyViewUtils.tsx [hunk 3]
```diff
@@ -2417,3 +2418,102 @@ export function getFilteredAndCompressedDataIntervalFilters(
...
+            clinicalData[sample.uniqueSampleKey] = {
+                ...(clinicalData[sample.uniqueSampleKey] || {}),
+                [item.clinicalAttributeId]: item.value,
+            };
+        });
+    });
+    return clinicalData;
+}
+
+export function updateChartIds(
```
Comment (alisman) [inline src/pages/studyView/StudyViewUtils.tsx]: i think this name should be more descriptive

**Pull request merged**